### DISCUSSION TITLE
Decrease loading time for scenes and homepage like IEEEVR

### DISCUSSION THREAD

**Discussion by jenn1991** [mozilla/hubs#2970] [Hubs Cloud Questions] [answered]
Hello, I have a question in cloud hubs, how can I make a page with the rooms created where I can see how many users are entering, I saw this in the IEEEVR experience but it is no longer available. Likewise, how can I improve the slow loading of 3d objects when it is in the cloud and is it feasible to place more than 25 people per room when it is in the cloud?

**Comment by DoisKoh** [comment, marked as answer]
Hello, I have this same exact question - and is this only possible via accessing that data from customizing the project using Hubs Cloud or is there a way to access this via some API from Mozilla Hubs?

**Reply by robinkwilson** [reply]
**Re: API**
We're doing a lot of work on the API, here's the WIP API PR: https://github.com/mozilla/reticulum/pull/399

Here's a gist from a community member, Utopiah, on how to get the user count:
https://gist.github.com/Utopiah/8bf9375ef73d46b6c471c3ec2abae405

**Re: Slow loading 3D objects**
The Hubs Cloud level generally isn't the bottleneck to slow loading assets (though you can up the instance size to **ensure** that isn't the bottleneck). 

The bottleneck is generally the user's internet connection which we can't control. However, we've added a medium quality mode you can toggle on for performance in the scene (the assets still need to load though).
https://github.com/mozilla/hubs/pull/2893
https://github.com/mozilla/hubs-cloud/blob/master/CHANGELOG.md#medium-quality-mode

**Ultimately, decrease the download size as much as possible.** 
Decrease the number of 3D objects in the scene + optimize the 3D objects via regular 3D asset optimizations: baking, decrease poly count, decrease texture size. 
When you save your scene in Spoke, the Hubs scene editor, it gives you a list of how well your scene is doing on the optimization standpoint:

**Reply by DoisKoh** [reply]
Hey, thanks a lot for the help!

**Reply by bemyevent** [reply]
Hi robin, do you have a sample of how to include the code in a custom hub cloud page ? Utopiah s code . Thx

**Reply by robinkwilson** [reply]
I currently do not. But I'm sure if you asked on the Hubs discord, someone in the community might!

**Reply by DoisKoh** [reply]
/api/v1/media/search?source=rooms&filter=public

I get internal server error whenever I try a different argument for the filter parameter. I can't get this for private rooms? I presume it's a permissions issue because then anyone can query on the status of private rooms? How can I get around that?
